codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxySSLTLV.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.codec.haproxy;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static io.netty.util.internal.ObjectUtil.*;
+
+/**
+ * Represents a {@link HAProxyTLV} of the type {@link HAProxyTLV.Type#PP2_TYPE_SSL}.
+ * This TLV encapsulates other TLVs and has additional information like verification information and a client bitfield.
+ */
+public final class HAProxySSLTLV extends HAProxyTLV {
+
+    private final int verify;
+    private final List<HAProxyTLV> tlvs;
+    private final Set<Client> clients;
+
+    /**
+     * The possible values of the clientfield bitmask
+     */
+    public enum Client {
+        PP2_CLIENT_SSL,
+        PP2_CLIENT_CERT_CONN,
+        PP2_CLIENT_CERT_SESS;
+    }
+
+    /**
+     * Creates a new HAProxySSLTLV
+     *
+     * @param verify the verification result
+     * @param clientBitField the bitfield with {@link Client} information
+     * @param tlvs the encapsulated {@link HAProxyTLV}s
+     * @param rawContent the raw TLV content
+     */
+    HAProxySSLTLV(final int verify, final byte clientBitField, final List<HAProxyTLV> tlvs, final byte[] rawContent) {
```

Comment:
is the `verify` terminology defined in a specification somewhere? If not can we provide a bit more descriptive name (e.g. `verificationResult`)? If it is can you provide a link in the javadocs?